src/utils.c

```diff
@@ -179,7 +180,11 @@ pu_write_raw(const gchar *input_path,
     if (input_finfo == NULL)
         return FALSE;
 
-    input_size = g_file_info_get_size(input_finfo);
+    if (size == 0)
+        input_size = g_file_info_get_size(input_finfo);
+    else
+        input_size = size * device->sector_size;
+
```

Comment:
I think we should handle negative sizes to be valid as well here. `PedSector` is a typedef to `long long` and is thus a signed integer. E.g. `-1` should also use the file's size.